controllers/space/SpaceController.php

```diff
@@ -2,44 +2,67 @@
 
 namespace humhub\modules\smartVillage\controllers\space;
 
+use Firebase\JWT\JWT;
+use humhub\modules\rest\models\ConfigureForm;
 use humhub\modules\smartVillage\components\AuthBaseController;
 use humhub\modules\space\models\Space;
 use humhub\modules\rest\definitions\SpaceDefinitions;
+use humhub\modules\user\models\User;
 
 class SpaceController extends AuthBaseController
 {
     public function actionFind(){
         $results = [];
-        $query = Space::find()->where(['visibility' => Space::VISIBILITY_ALL]);
+
+        //Check the requested user is guest or not
+        $user = $this->checkUserIsRegistered();
+        if($user == false){
+            $query = Space::find()->where(['visibility' => Space::VISIBILITY_ALL]);
+        }else{
+            $query = Space::find()->where(['visibility' => Space::VISIBILITY_REGISTERED_ONLY]);
+
+        }
```

Comment:
now if there is no user, all spaces with `"visibility": 2` are returned and if there is a user only the spaces with `"visibility": 1,`. but if the user is logged in there should be all spaces returned, 1s and 2s, you know? can you please adjust this as well @gaurav-handysolver.

Reply:
when i am thinking about different cases i guess we need every space except the ones with `"visibility": 0` in the second case with a user, right?

Reply:
the three possible states: https://github.com/humhub/humhub/blob/10c8ee6702b1ab1196db6cb7366b3ed18e003493/protected/humhub/modules/space/models/Space.php#L76-L78

so we need "where not NONE" right?

Reply:
Done